rpc/src/rpc.rs

```diff
@@ -510,6 +510,62 @@ impl JsonRpcRequestProcessor {
         })
     }
 
+    pub fn get_secondary_index_key_size (
+        &self,
+        index_key: &Pubkey,
+        commitment: Option<CommitmentConfig>
+    ) -> Result<RpcResponse<Option<HashMap<JsonAccountIndex, usize>>>>
+    {        
+        // Grab RPC server's secondary index config
+        // Exit if secondary indexes are not enabled
+        let secondary_index_config = 
+            if !self.config.account_indexes.is_empty() {
+                &self.config.account_indexes
+            }
+            else { return None.ok_or(Error::from(RpcCustomError::SecondaryIndexDoesNotExist)); };
+        
+        // Make sure your key is not explicitly excluded
+        if !self.config.account_indexes.include_key(index_key) {
+            return None.ok_or(Error::from(RpcCustomError::KeyExcludedFromSecondaryIndex {
+                index_key: index_key.to_string(),
+            }));
+        }
+
+        // Aquire the bank
+        let bank = self.bank(commitment);
+
+        // Grab a ref to the AccountsIndex for this Bank
+        let accounts_index = &bank.accounts().accounts_db.accounts_index;
+
+        // Find the size of the key in every index where it exists
+        let found_sizes = 
+            Some(secondary_index_config
+                .indexes
+                .iter()
+                .filter_map(|idx| 
+                    {
+                        let sz = accounts_index.get_index_key_size(idx, index_key);
+                        (sz.is_some())
+                        .then(|| 
+                        (match idx {
+                            AccountIndex::ProgramId => JsonAccountIndex::ProgramId,
+                            AccountIndex::SplTokenOwner => JsonAccountIndex::SplTokenOwner,
+                            AccountIndex::SplTokenMint => JsonAccountIndex::SplTokenMint
```

Comment:
How about implementing some kind of conversion method to simplify this code? I expect we'll need it again for your later work. Unfortunately, it'll have to live here in solana-rpc, as opposed to implementing From on RpcAccountIndex, since we don't want to add a runtime dependency to the client-api crate, or an rpc dependency to runtime.

Reply:
Addressed